alhimick-prog/hometask-3/rainfall.rb

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+INPUT_PROMPT = 'Enter city name:'
+ERROR_BLANK_STRING = 'City name can not be blank!'
+ERROR_FILE_READ_FAIL = 'Error: data file not found'
+MESSAGE_INCORRECT_MONTH_NUMBER = 'Warning: the number of months is not equal to 12.'
+MEAN = 'Rainfall mean:'
+VARIANCE = 'Rainfall variance:'
+EXIT_STRING = 'exit!'
+TOWN_SEARCH_ERROR_NUMBER = -1
+MONTH_NUMBER = 12
+CURRENT_FILE = 'data.txt'
+
+def file_reader(filename)
+  if File.exist?(filename)
+    file = File.new(filename, 'r')
+    file.read(file.size)
+  else
+    puts ERROR_FILE_READ_FAIL
+    exit
+  end
+end
+
+def form_array(town, strng)
+  array = nil
```

Comment:
@alhimick-prog `array = []`

Reply:
@lincAlinc what if the string with the word exists but has no meaning? How do we distinguish it from a negative search result?